error[E0603]: trait `StrExt1` is private
  --> $DIR/visibility-2.rs:34:14
   |
34 |     use foo::StrExt1; //~ ERROR trait `StrExt1` is private [E0603]
   |              ^^^^^^^ private trait
   |
note: the trait `StrExt1` is defined here
  --> $DIR/visibility-2.rs:4:5
   |
4  |     #[ext(StrExt1)]
   |     ^^^^^^^^^^^^^^^
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0603]: trait `StrExt2` is private
  --> $DIR/visibility-2.rs:36:14
   |
36 |     use foo::StrExt2; //~ ERROR trait `StrExt2` is private [E0603]
   |              ^^^^^^^ private trait
   |
note: the trait `StrExt2` is defined here
  --> $DIR/visibility-2.rs:11:5
   |
11 |     #[ext(pub(self) StrExt2)]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0603]: trait `StrExt3` is private
  --> $DIR/visibility-2.rs:38:19
   |
38 |     use foo::bar::StrExt3; //~ ERROR trait `StrExt2` is private [E0603]
   |                   ^^^^^^^ private trait
   |
note: the trait `StrExt3` is defined here
  --> $DIR/visibility-2.rs:21:9
   |
21 |         #[ext(pub(super) StrExt3)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
